tests/functionality_tests/test_activation_shapes.py

```diff
@@ -25,15 +52,15 @@ def main():
         rec_col_limit  = 2 * cols // 3## Good value 22
         grid.activate_rectangle(rec_row_start, rec_col_start, rec_row_limit, rec_col_limit)
         grid.find_border_nodes()
-        way_points = grid.get_all_lawnmower_waypoints_adjustable()
+        way_points = grid.get_all_guided_lawnmower_waypoints_adjustable(is_vertical)
 
     if activation_type== "circle":
         circle_center_row = rows // 2 ## Good value 19
         circle_center_col = cols // 2 ## Good value 17
         circle_radius = min(rows, cols) // 2 ## Good value 17
         grid.activate_circle(circle_center_row, circle_center_col, circle_radius)  
         grid.find_border_nodes()
-        way_points = grid.get_all_lawnmower_waypoints_adjustable()
+        way_points = grid.get_all_guided_lawnmower_waypoints_adjustable(is_vertical)
 
     if activation_type== "triangle":
         x1 = rows // 6## Good value 6
```

Comment:
```suggestion
    # TODO: Vertical traversal in triangle
    if activation_type== "triangle":
        x1 = rows // 6## Good value 6
```
Let's make a note that triangle vertical traversal doesn't work

Reply:
Yes, but we did talk about this last meeting. The vertical traversal currently does not work because of the weird orientation. But we could make a separate triangle test for vertical traversal.

Reply:
Oh, then let's either mention that its for the specific orientations you found or change this test script to use a case that works (bc it could be confusing to someone why triangle doesn't have a line in it when we run the script)

Reply:
actually both would prob be best